internal/service/quicksight/data_set.go

```diff
@@ -0,0 +1,2800 @@
+			},
+
+			"logical_table_map": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Computed: true,
+				MaxItems: 1,
+				// key length constraint 1 to 64
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"alias": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringLenBetween(1, 64),
+						},
+						"source": {
+							Type:     schema.TypeList,
+							Required: true,
+							MaxItems: 1,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"data_set_arn": {
+										Type:     schema.TypeString,
+										Computed: true,
+										Optional: true,
+									},
+									"join_instruction": {
+										Type:     schema.TypeList,
+										Computed: true,
+										Optional: true,
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"left_operand": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validation.StringLenBetween(1, 64),
+												},
+												"on_clause": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validation.StringLenBetween(1, 512),
+												},
+												"right_operand": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validation.StringLenBetween(1, 64),
+												},
+												"type": {
+													Type:         schema.TypeString,
+													Required:     true,
+													ValidateFunc: validation.StringInSlice(quicksight.JoinType_Values(), false),
+												},
+												"left_join_key_properties": {
+													Type:     schema.TypeList,
+													Computed: true,
+													Optional: true,
+													MaxItems: 1,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"unique_key": {
+																Type:     schema.TypeBool,
+																Computed: true,
+																Optional: true,
+															},
+														},
+													},
+												},
+												"right_join_key_properties": {
+													Type:     schema.TypeList,
+													Computed: true,
+													Optional: true,
+													MaxItems: 1,
+													Elem: &schema.Resource{
+														Schema: map[string]*schema.Schema{
+															"unique_key": {
+																Type:     schema.TypeBool,
+																Computed: true,
+																Optional: true,
+															},
+														},
+													},
+												},
+											},
+										},
+									},
+									"physical_table_id": {
+										Type:         schema.TypeString,
+										Computed:     true,
+										Optional:     true,
+										ValidateFunc: validation.StringLenBetween(1, 64),
+									},
+								},
+							},
```

Comment:
Added to commit: "review-1"

Reply:
I tried running the provider with this included and I'm running into this error:

`resource aws_quicksight_data_set: ExactlyOneOf: data_set_arn references unknown attribute (data_set_arn) at part (data_set_arn)`

I looked at how other files implement `ExactlyOneOf:` but I can't figure out what I'm doing wrong here. At first I thought it was the `Computed: true` but I found a couple other files that include that. Any thoughts?

Reply:
I believe this is because we need to specify the particular config block in which the nested attribute appears -- in this case: `logical_table_map.0.source.data_set_arn`. Changing the `ExactlyOneOf` line to the following seems to fix:
```
ExactlyOneOf: []string{"logical_table_map.0.source.data_set_arn", "logical_table_map.0.source.join_instruction", "logical_table_map.0.source.physical_table_id"},
```
An example of this kind of reference can be seen in the `user_pool_client` resource for `cognitoidp` ([code in GitHub](https://github.com/hashicorp/terraform-provider-aws/blob/a3ae79cc263017afba6d2c536c7dad4838fa14b5/internal/service/cognitoidp/user_pool_client.go#L72))

Reply:
I believe that this is because we need to explicitly reference the block within which the attribute is nested -- i.e. in this case: `logical_table_map.0.source.data_set_arn`.

Changing the `ExactlyOneOf` line to the following ought to fix:
```
ExactlyOneOf: []string{"logical_table_map.0.source.data_set_arn", "logical_table_map.0.source.join_instruction", "logical_table_map.0.source.physical_table_id"},
```